From 2732d5954ccf2449fdaecb1467d7b4223cae797b Mon Sep 17 00:00:00 2001
From: Pierre-Yves MORDRET <pierre-yves.mordret@foss.st.com>
Date: Tue, 13 Aug 2024 17:38:17 +0200
Subject: [PATCH] arm64: dts: Add GPU Thermal trip points

Change-Id: I2a39acb0e1271083a59f1f2b1c7841f6460b7158
Signed-off-by: Pierre-Yves MORDRET <pierre-yves.mordret@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/402785
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Philippe CORNU <philippe.cornu@foss.st.com>
Reviewed-by: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
Domain-Review: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
---
 arch/arm64/boot/dts/st/stm32mp255.dtsi | 24 ++++++++++++++++++++++++
 1 file changed, 24 insertions(+)

--- a/arch/arm64/boot/dts/st/stm32mp255.dtsi
+++ b/arch/arm64/boot/dts/st/stm32mp255.dtsi
@@ -6,6 +6,27 @@
 #include <dt-bindings/power/st,stm32mp25-power.h>
 #include "stm32mp253.dtsi"
 
+/ {
+	thermal-zones {
+		cpu-thermal {
+			trips {
+				gpu_alert: gpu-alert {
+					temperature = <110000>;
+					hysteresis = <10000>;
+					type = "passive";
+				};
+			};
+
+			cooling-maps {
+				map1 {
+					trip = <&gpu_alert>;
+					cooling-device = <&gpu 1 6>;
+				};
+			};
+		};
+	};
+};
+
 &ltdc {
 	clocks = <&rcc CK_BUS_LTDC>, <&rcc CK_KER_LTDC>, <&syscfg 0>, <&lvds 0>;
 	clock-names = "bus", "ref", "lcd", "lvds";
@@ -68,5 +89,8 @@
 		power-domains = <&scmi_devpd PD_SCMI_GPU>, <&CLUSTER_PD>;
 		access-controllers = <&rifsc 79>;
 		status = "disabled";
+
+		throttle,max_state = <6>;
+		#cooling-cells = <2>;
 	};
 };
